enzy_htp/structure/structure_region/api.py

```diff
@@ -61,10 +63,25 @@ def atoms(self) -> List[Atom]:
         return self.atoms_
 
     @atoms.setter
-    def atoms(self, val: List[Atom]):
+    def atoms(self, val: List[Atom]) -> None:
         """setter for atoms_"""
         self.atoms_ = val
 
+        for atom in self.atoms:
+            if atom.parent.parent is None:
+                atom.parent.parent = self
```

Comment:
We should not set the parent here. In this case, it will lose track to the Structure() context (i.e.: the self.topology will no longer work)

Reply:
We need this context in many places especially when dealing with ElectronicStrutcure.

Reply:
I can tweak the code here a bit. In my experience this only applies to `ResidueCap` since they are not owned by the `Structure`. Let me know what to do.

Reply:
Oh yes it only for the later comers after ctor. But still, ResidueCap atoms should also not be parented by the region but by the cap itself. see another related comment above.

Reply:
You are correct, and I think it is best for the `Atom()`'s in the `ResidueCap()` to be owned by the Cap and not the `StructureRegion`. I removed the above code that sets the `atom.parent.parent` to not `None`.